tastyworks/tastyworks_api/tw_api.py

```diff
@@ -0,0 +1,647 @@
+"""
+
+
+async def session_start(username: str = None, password: str = None) -> Dict:
+    """
+    Start a TW API session.
+
+    Args:
+        username: TW account username
+        password: TW account password
+
+    Returns:
+        The API server response containing the session token as well as user/account data
+    """
+    if username is None and password is None:
+        body = {
+            'login': environ.get('TW_USER', ""),
+            'password': environ.get('TW_PASSWORD', "")
+        }
+    else:
+        body = {
+            'login': username,
+            'password': password
+        }
+
+    url = make_api_url('SESSION_START')
+    resp = await api_request('POST', url, json_data=body)
+    return resp
+
+
+async def session_validate(token: str) -> Dict:
+    """
+    Validate the status of a TW API session using the session token
+    """
+    url = make_api_url('SESSION_VALIDATE')
+    resp = await api_request('POST', url, token)
+    return resp
+
+
+async def get_streamer_info(token: str) -> Dict:
+    """
+    Get the streamer information including URL and token (take in the different TW session token)
+    """
+    url = make_api_url('STREAMER_INFO')
+    resp = await api_request('GET', url, token)
+    return resp
+
+
+"""
+#########################################
+############### ACCOUNT #################
+#########################################
+"""
+
+
+async def get_accounts(token: str) -> Dict:
+    """
+    Requests the account numbers using the session token
+    """
+    url = make_api_url('ACCOUNTS')
+    resp = await api_request('GET', url, token)
+    return resp
+
+
+async def get_balances(token: str, account_number: str) -> Dict:
+    """
+    Requests the balances for a specific account number using the session token
+    """
+    url = make_api_url('BALANCES', account=account_number)
+    resp = await api_request('GET', url, token)
+    return resp
+
+
+async def get_positions(token: str, account_number: str) -> Dict:
+    """
+    Retrieves all the positions for a specific account number using the session token
+    """
+    url = make_api_url('POSITIONS', account=account_number)
+    resp = await api_request('GET', url, token)
+    return resp
+
+
+async def get_status(token: str, account_number: str) -> Dict:
+    """
+    Retrieves the status of an account
+    """
+    url = make_api_url('STATUS', account=account_number)
+    resp = await api_request('GET', url, token)
+    return resp
+
+
+async def get_capital_req(token: str, account_number: str) -> Dict:
+    """
+    Retrieves the status of an account
+    """
+    url = make_api_url('CAPITAL_REQ', account=account_number)
+    resp = await api_request('GET', url, token)
+    return resp
+
+"""
+#########################################
+######## TRANSACTIONS & ORDERS ##########
+#########################################
+"""
+
+
+async def get_transactions(token: str, account_number: str = '', symbol: str = '',
```

Comment:
I believe this is handled by `Tastyworks.models.trading_account.TradingAccount.get_history()`, but lacking pagination (waiting on PR)